### PR TITLE
88731 - Adds new VideoLink component to appointment details redesign video details page

### PR DESCRIPTION
**Pull request by ryanshaw**
## Summary

- Added new separate `VideoLink` component for the appointment details redesign. When we are confident with the redesign and it's been deployed to 100% of production users, it will be easier to remove the old component.
- Added new unit tests for the new component

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/88731

## Testing done

- Local testing
- Unit testing

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |Screenshot 2024-08-01 at 06-13-45 VA appointment on Sunday September 22 2024 Veterans Affairs|Screenshot 2024-08-01 at 06-14-47 VA appointment on Sunday September 22 2024 Veterans Affairs|
| Desktop |Screenshot 2024-08-01 at 06-13-14 VA appointment on Sunday September 22 2024 Veterans Affairs|Screenshot 2024-08-01 at 06-14-34 VA appointment on Sunday September 22 2024 Veterans Affairs|

## Acceptance criteria
Background: the `va_online_scheduling_appointment_details_redesign` toggle is on
- [ ] Given when it is 30 minutes before an at home video appointment or 4 hours after the start time
- [ ] Then the front end will display the _Join appointment_ button
- [ ] And page design matches the [design spec](https://www.figma.com/design/eonNJsp57eqfPqx7ydsJY9/Feature-Reference-%7C-Appointments-FE?node-id=756-20952&t=IjUEbMHzkxesIHMX-4)
### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/vaos/components/tests/VideoLink.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+beforeEach(() => {
+  MockDate.set(getTestDate());
+});
+
+afterEach(() => {
+  MockDate.reset();
+});
+
+describe('VAOS Component: VideoLink', () => {
+  it('does not render join appoinment link when not 30 minutes before or 4 hours after start time', () => {
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/tests/VideoLink.unit.spec.js:19]: Let's split this test into two cases, before 30 mins and after 4 hours.

File: src/applications/vaos/components/VideoLink.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+// import classNames from 'classnames';
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/VideoLink.jsx:2]: nit: remove unused imports.

File: src/applications/vaos/components/VideoLink.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+export default function VideoLink({ appointment }) {
+  const { url } = appointment.videoData;
+  const diff = moment().diff(moment(appointment.start), 'minutes');
+
+  // Button is enabled 30 minutes prior to start time, until 4 hours after start time
+  // NOTE: If the moment is earlier than the moment you are passing to moment.fn.diff,
+  // the return value will be negative. So checking to see if the appointment start
+  // time is before or after the current time.
+  let disableVideoLink;
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/VideoLink.jsx:15]: This seems too verbose. How about something like:
```
const disableVideoLink = diff > 30 || diff < -240;
```

**Review by JunTaoLuo — COMMENTED** [review]
A few small questions

**Comment by ryanshaw** [comment]
@JunTaoLuo Thanks for the quick review, just pushed changes.

**Review by JunTaoLuo — APPROVED** [review]
LGTM

**Comment by JunTaoLuo** [comment]
I can't access the figma file so I can't check the UI against the mockup but the code changes look good to me.

**Pull request merged**